dan_layer/core/src/workers/hotstuff_waiter.rs

```diff
@@ -686,19 +694,16 @@ where
 
         // TODO: Fix merkle proof verification
         // all merkle proofs for the signers must be valid
-        // let validator_node_root = self.epoch_manager.get_validator_node_merkle_root(qc.epoch()).await?;
-        // for md in qc.validators_metadata() {
-        //     let merkle_proof: MerkleProof = bincode::deserialize(&md.merkle_proof).map_err(|_| {
-        //         HotStuffError::InvalidQuorumCertificate("could not deserialize merkle proof".to_string())
-        //     })?;
-        //     merkle_proof
-        //         .verify_leaf::<ValidatorNodeMmrHasherBlake256>(
-        //             &validator_node_root,
-        //             &md.public_key,
-        //             md.merkle_leaf_index as usize,
-        //         )
-        //         .map_err(|_| HotStuffError::InvalidQuorumCertificate("invalid merkle proof".to_string()))?;
-        // }
+        let validator_node_root = self.epoch_manager.get_validator_node_merkle_root(qc.epoch()).await?;
+        for md in qc.validators_metadata() {
```

Comment:
Another future optimization: These merkleproofs can be combined into a single merkle proof before the QC is constructed. Then only one validation is needed